packages/shared/scope-hoisting/src/formats/global.js

```diff
@@ -97,7 +101,7 @@ export function generateExports(
     exported.add(exportsId);
 
     statements.push(
-      EXPORT_TEMPLATE({
+      EXPORT_FN_TEMPLATE({
```

Comment:
Not sure I understand this change. Could you explain?

What about other output formats, e.g. ESM/CommonJS?

Reply:
With esm and commonjs, we directly import from the bundle, but with the global one, we register the exports via `parcelRequire.register(id, exports)`.
If a bundle has the asset twice in it's ancestry, both as a child bundle and as a shared bundle, this happened:
```js
// parent bundle 1 (child, 
// would naively not need a init because it is initialized by being loaded at runtime)
parcelRequire.register("1234", $1234$exports)
```
```js
// parent bundle 2 (sibling)
parcelRequire.register("1234", $1234$init)
```
Now we got a collision and don't know if we need to do `parcelRequire("1234")()` or not.

(Thinking about it, it actually does make a difference which of these you get because some state could be set in the asset - but that is another issue).

I hope I got this right, @wbinnssmith.

Reply:
Ah ok, that makes sense.

Reply:
Could we add a comment to the code explaining this?

Reply:
done 😄